numpyro/primitives.py

```diff
@@ -131,6 +131,29 @@ def sample(
         + "_unobserved"`` which should be used by guides.
     :return: sample from the stochastic `fn`.
     """
+    if not isinstance(fn, numpyro.distributions.Distribution):
+        type_error = TypeError(
+            "It looks like you tried to use a fn that isn't an instance of "
+            "numpyro.distributions.Distribution, or "
+            "tensorflow_probability.distributions.Distribution. If you're using "
+            "tensorflow_probability, make sure it is correctly installed."
+        )
+        try:
+            from tensorflow_probability.substrates.jax import distributions as tfd
+
+            from numpyro.contrib.tfp.distributions import TFPDistribution
+        except ImportError:
+            # if tensorflow_probability fails to import, then fn can't have been a tf
+            # distribution in the first place, so returning TypeError is ok
+            raise type_error
```

Comment:
Are you happy with this reasoning? Error message includes a note to make sure tfp is installed correctly if the user is trying to use it but is still getting an `ImportError`.

Reply:
I like this pattern. :+1: